### ISSUE TITLE
[analyzer] FN: null deref gets suppressed without defensive checks

### ISSUE THREAD

**Issue by rikki322** [llvm/llvm-project#213552]
Hi, I found a false negative in `core.CallAndMessage`.

CSA misses a null C++ receiver when the null value is returned by a trivial helper.

## Program

```cpp
class TestInstanceCall {
public:
  void foo() {}
};

TestInstanceCall *get_null() { return 0; }

void test_ic_null() {
  TestInstanceCall *p = get_null();
  p->foo();
}
```

The receiver `p` is null at the call to `p->foo()`, so this is the same bug pattern as a direct `TestInstanceCall *p = 0; p->foo();`.

## Actual result

Re-running CSA on the reproducer emits no `core.CallAndMessage` result:

```text
No results.
```

No warning is emitted for the member call:

```cpp
p->foo();
```

## Expected result

CSA should report `core.CallAndMessage` for the member call through `p`, because `get_null()` always returns null.

## Verification

```sh
clang++ --analyze \
  -Xclang -analyzer-checker=core.CallAndMessage \
  -Xclang -analyzer-output=sarif \
  -o callandmessage_get_null_receiver_fn.sarif \
  callandmessage_get_null_receiver_fn.cpp
```

Version:

```text
clang version 24.0.0git (https://github.com/llvm/llvm-project.git 1cb7e838cd47ecad4050948c0c907ecb1f466ac3)
```

**Comment by rikki322** [comment]
Thanks for reviewing this issue.

I noticed it is labeled as `false-positive`. Just to clarify, this report is about a missing diagnostic rather than an incorrect diagnostic. Since CSA does not emit a warning for the null receiver case, I believe it should be classified as a false negative.

Could you please take another look at the label?

**Comment by steakhal** [comment]
This is the consequence of the so-called inlined defensive checks suppression heuristic.
You can override it by setting `-analyzer-config suppress-null-return-paths=false`, after which you would see the bug report appear.
https://godbolt.org/z/1P8v6hEjv

This works as intended - albeit stresses the point that this suppression heuristic isn't really great.
